physlr/physlr.py

```diff
@@ -1021,107 +1020,118 @@ def assign_read_molecule(mxs, moltomxs, mol_counts, bx):
-                communities += list(nxcommunity.k_clique_communities(g.subgraph(comp), 3))
-        return u, {v: i for i, vs in enumerate(communities) if len(vs) > 1 for v in vs}
+        return components
 
     @staticmethod
-    def determine_molecules_louvain(g, u):
-        "Apply louvain community detection algorithm after extracting bi-connected components."
-        import community as louvain
+    def detect_communities_k_clique(g, node_set, k=3):
+        """Apply k-clique community detection. Return communities."""
+        from networkx.algorithms import community as nxcommunity
 
-        cut_vertices = set(nx.articulation_points(g.subgraph(g.neighbors(u))))
-        components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)) - cut_vertices)))
-        components.sort(key=len, reverse=True)
-        communities = []
-        for comp in components:
-            if len(comp) > 1:
-                partition = louvain.best_partition(g.subgraph(comp))
-                for com in set(partition.values()):
-                    list_nodes = [nodes for nodes in partition.keys() if partition[nodes] == com]
-                    if len(list_nodes) > 1:
-                        communities.append(list_nodes)
-        return u, {v: i for i, vs in enumerate(communities) if len(vs) > 1 for v in vs}
+        if len(node_set) > 1:
+            return [set(i) for i in nxcommunity.k_clique_communities(g.subgraph(node_set), k)]
+        return []
 
     @staticmethod
-    def determine_molecules_just_louvain(g, u):
-        "Apply louvain community detection without bi-connected separation."
+    def detect_communities_louvain(g, node_set, init_communities=False):
+        """Apply Louvain community detection on a single component. Return communities."""
         import community as louvain
 
-        sub_graph = g.subgraph(g.neighbors(u))
-        nodes_count = len(sub_graph)
-        if nodes_count == 0:  # or edges_count == 0:
-            components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)))))
-            return u, {v: i for i, vs in enumerate(components) if len(vs) > 1 for v in vs}
-        partition = louvain.best_partition(sub_graph)
-        if not partition:
-            return u, {}
-        multinode_partitions_set = {}
-        for com in set(partition.values()):
-            list_nodes = [nodes for nodes in partition.keys() if partition[nodes] == com]
-            if len(list_nodes) > 1:
-                multinode_partition = {v: com for v in list_nodes}
-                multinode_partitions_set.update(multinode_partition)
-        return u, multinode_partitions_set
+        if len(node_set) > 1:
+            if not init_communities:
+                partition = louvain.best_partition(g.subgraph(node_set))
+            else:
+                partition = louvain.best_partition(g.subgraph(node_set), init_communities)
+            return [{nodes for nodes in partition.keys() if partition[nodes] == com}
```

Comment:
```suggestion
            return [{u for u in partition.keys() if partition[u] == com}
```
`nodes` is I believe a single node, yes? If so, it the variable name should be `node` or `u` rather than `nodes`.

Reply:
Right. changed.